R/precompiled.R

```diff
@@ -0,0 +1,54 @@
+
+# This figures out which precompiled CSS files are available.
+# This is memoized - it runs once per R session.
+precompiled_css <- local({
+  themes <- NULL
+
+  function() {
+    if (is.null(themes)) {
+      versions <- dir(system.file("css-precompiled", package = "bootstraplib"))
+      themes <<- lapply(versions, function(version) {
+        list(
+          version = version,
+          theme = bs_theme_create(version)
+        )
+      })
+    }
+    themes
+  }
+})
+
+
+precompiled_css_version <- function(theme) {
+  for (precompiled in precompiled_css()) {
+    if (identical(theme, precompiled$theme)) {
+      return(theme_version(theme))
+    }
+  }
+
+  return(NULL)
+}
+
+
+#' Get the path to a precompiled CSS file
+#'
+#' @param theme A theme object.
+#'
+#' @return The path to the precompiled CSS file, if it exists, or `NULL`, if
+#'   not.
+#'
+#' @examples
+#' # The path to a precompiled CSS file for a theme
+#' precompiled_css_path(bs_theme_create("4"))
+#'
+#' # Attempting to get the path to a theme that is not precompiled will return NULL
+#' precompiled_css_path(bs_theme_create(bootswatch = "sketchy"))
+#' @export
```

Comment:
I feel uneasy about exporting this...if we give them this path, they can easily shoot themselves in the foot by not considering the `file_attachments`

Reply:
I wanted to have some way that people could check if there was a precompiled versin of their theme. Without this, there's no way for users to check. Any ideas on another way to do it?

Reply:
I may be missing something, but why should users care? If they really do, they can sniff the `src` path of the returned `htmlDependency()`

Reply:
How about making it `@keywords internal`?

Reply:
I'd prefer un-exported until there's a request/need for it, but yes, it should at least be `@keywords internal`

Reply:
Also, if we do export, we should add a disclaimer to the `@description`